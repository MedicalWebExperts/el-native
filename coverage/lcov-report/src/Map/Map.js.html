<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for src/Map/Map.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../prettify.css" />
    <link rel="stylesheet" href="../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../index.html">All files</a> / <a href="index.html">src/Map</a> Map.js
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">62.07% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>18/29</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">83.33% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>10/12</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">37.5% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>3/8</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">66.67% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>18/27</span>
      </div>
    </div>
  </div>
  <div class='status-line medium'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41
42
43
44
45
46
47
48
49
50
51
52
53
54
55
56
57
58
59
60
61
62
63
64
65
66
67
68
69
70
71
72
73
74
75
76
77
78
79
80
81
82
83
84
85
86
87
88
89
90
91
92
93
94
95
96
97
98
99
100
101
102
103
104
105
106
107
108
109
110
111
112
113
114
115
116
117
118
119
120
121
122
123
124
125
126
127
128
129
130</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">5x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">5x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">5x</span>
<span class="cline-any cline-yes">5x</span>
<span class="cline-any cline-yes">5x</span>
<span class="cline-any cline-yes">5x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">5x</span>
<span class="cline-any cline-yes">5x</span>
<span class="cline-any cline-yes">5x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">5x</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">5x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">5x</span>
<span class="cline-any cline-yes">5x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">5x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">/* eslint no-console: ["error", { allow: ["error"] }] */
import React from 'react';
import PropTypes from 'prop-types';
import qs from 'qs';
import {
  View,
  Image,
  Text,
  Dimensions,
  Platform,
  TouchableHighlight,
  Linking,
} from 'react-native';
&nbsp;
import Styles from './styles';
&nbsp;
const propTypes = {
  /**
   * Description of prop "latitude".
   */
  latitude: PropTypes.number.isRequired,
  /**
   * Description of prop "longitude".
   */
  longitude: PropTypes.number.isRequired,
  /**
   * Description of prop "width".
   */
  width: PropTypes.number,
  /**
   * Description of prop "height".
   */
  height: PropTypes.number,
  /**
   * Description of prop "apiKey".
   */
  apiKey: PropTypes.string.isRequired,
  maptype: PropTypes.string,
  zoom: PropTypes.number, // eslint-disable-line react/no-unused-prop-types
  onPress: PropTypes.func,
  style: PropTypes.object,
  button: PropTypes.element, // eslint-disable-line react/no-unused-prop-types
  underlayColor: PropTypes.string, // eslint-disable-line react/no-unused-prop-types
};
&nbsp;
const defaultProps = {
  width: undefined,
  height: undefined,
  maptype: 'roadmap',
  zoom: 18,
  onPress: undefined,
  style: {},
  button: undefined,
  underlayColor: 'white',
};
&nbsp;
/**
 * Map component description.
 */
export default function Map(props) {
  const dims = Dimensions.get('window');
&nbsp;
  const computedDimensions = {
    width: Math.round(props.width || dims.width),
    height: Math.round(props.height || dims.width * (2 / 3)),
  };
&nbsp;
  const googleStaticMapUri = () =&gt; {
    const baseUrl = 'https://maps.googleapis.com/maps/api/staticmap';
    const apiKey = props.apiKey;
    const params = {
      markers: [props.latitude, props.longitude].join(','),
      maptype: props.maptype,
      size: [computedDimensions.width, computedDimensions.height].join('x'),
      scale: 2,
      apiKey,
    };
    const query = qs.stringify(params, { arrayFormat: 'brackets' });
    const url = `${baseUrl}?${query}`;
    return url;
  };
&nbsp;
  const handleOnPress = <span class="fstat-no" title="function not covered" >()</span> =&gt; {
    const generateIos = <span class="cstat-no" title="statement not covered" ><span class="fstat-no" title="function not covered" >()</span> =&gt; <span class="cstat-no" title="statement not covered" >`https://maps.apple.com/?q=${props.latitude},${props.longitude}&amp;z=${props.zoom}`;</span></span>
    const generateAndroid = <span class="cstat-no" title="statement not covered" ><span class="fstat-no" title="function not covered" >()</span> =&gt; <span class="cstat-no" title="statement not covered" >`geo:${props.latitude},${props.longitude}`;</span></span>
&nbsp;
    const url = <span class="cstat-no" title="statement not covered" >Platform.select({</span>
      ios: generateIos,
      android: generateAndroid,
    })();
&nbsp;
<span class="cstat-no" title="statement not covered" >    Linking.canOpenURL(url).then(<span class="fstat-no" title="function not covered" >(s</span>upported) =&gt; {</span>
<span class="cstat-no" title="statement not covered" >      if (!supported) {</span>
<span class="cstat-no" title="statement not covered" >        console.error(`Can't handle url: ${url}`);</span>
<span class="cstat-no" title="statement not covered" >        return;</span>
      }
<span class="cstat-no" title="statement not covered" >      Linking.openURL(url);</span>
    }).catch(<span class="fstat-no" title="function not covered" >er</span>r =&gt; <span class="cstat-no" title="statement not covered" >console.error('An error occurred', err))</span>;
  };
&nbsp;
  const onPress = (props.onPress &amp;&amp; typeof props.onPress === 'function') ? props.onPress : handleOnPress;
&nbsp;
  const renderButton = () =&gt; (
    &lt;TouchableHighlight
      onPress={onPress}
      underlayColor={props.underlayColor}
      style={{ ...Styles.button, ...props.style.button }}
    &gt;
      &lt;View&gt;
        {props.button || &lt;Text&gt;Navigate&lt;/Text&gt;}
      &lt;/View&gt;
    &lt;/TouchableHighlight&gt;
  );
&nbsp;
  return (
    &lt;View style={{ ...Styles.container, ...props.style.container }}&gt;
      &lt;Image
        source={{
          uri: googleStaticMapUri(),
        }}
        style={computedDimensions}
      /&gt;
      {renderButton()}
    &lt;/View&gt;
  );
}
&nbsp;
Map.propTypes = propTypes;
Map.defaultProps = defaultProps;
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Wed Jan 17 2018 12:58:56 GMT-0300 (-03)
</div>
</div>
<script src="../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../sorter.js"></script>
</body>
</html>
