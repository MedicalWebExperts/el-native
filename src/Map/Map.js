/* eslint no-console: ["error", { allow: ["error"] }] */
import React from 'react';
import PropTypes from 'prop-types';
import qs from 'qs';
import { View, Image, Text, Dimensions, Platform, TouchableHighlight, Linking } from 'react-native';

import Styles from './styles';

const propTypes = {
  /**
   * Description of prop "latitude".
   */
  latitude: PropTypes.number.isRequired,
  /**
   * Description of prop "longitude".
   */
  longitude: PropTypes.number.isRequired,
  /**
   * Description of prop "width".
   */
  width: PropTypes.number,
  /**
   * Description of prop "height".
   */
  height: PropTypes.number,
  /**
   * Description of prop "apiKey".
   */
  apiKey: PropTypes.string.isRequired,
  maptype: PropTypes.string,
  zoom: PropTypes.number, // eslint-disable-line react/no-unused-prop-types
  onPress: PropTypes.func,
  style: PropTypes.object,
  button: PropTypes.element, // eslint-disable-line react/no-unused-prop-types
  underlayColor: PropTypes.string, // eslint-disable-line react/no-unused-prop-types
};

const defaultProps = {
  width: undefined,
  height: undefined,
  maptype: 'roadmap',
  zoom: 18,
  onPress: undefined,
  style: {},
  button: undefined,
  underlayColor: 'white',
};

/**
 * Map component description.
 */
export default function Map(props) {
  const dims = Dimensions.get('window');

  const computedDimensions = {
    width: Math.round(props.width || dims.width),
    height: Math.round(props.height || dims.width * (2 / 3)),
  };

  const googleStaticMapUri = () => {
    const baseUrl = 'https://maps.googleapis.com/maps/api/staticmap';
    const { apiKey } = props;
    const params = {
      markers: [props.latitude, props.longitude].join(','),
      maptype: props.maptype,
      size: [computedDimensions.width, computedDimensions.height].join('x'),
      scale: 2,
      apiKey,
    };
    const query = qs.stringify(params, { arrayFormat: 'brackets' });
    const url = `${baseUrl}?${query}`;
    return url;
  };

  const handleOnPress = () => {
    const generateIos = () =>
      `https://maps.apple.com/?q=${props.latitude},${props.longitude}&z=${props.zoom}`;
    const generateAndroid = () => `geo:${props.latitude},${props.longitude}`;

    const url = Platform.select({
      ios: generateIos,
      android: generateAndroid,
    })();

    Linking.canOpenURL(url)
      .then((supported) => {
        if (!supported) {
          console.error(`Can't handle url: ${url}`);
          return;
        }
        Linking.openURL(url);
      })
      .catch(err => console.error('An error occurred', err));
  };

  const onPress =
    props.onPress && typeof props.onPress === 'function' ? props.onPress : handleOnPress;

  const renderButton = () => (
    <TouchableHighlight
      onPress={onPress}
      underlayColor={props.underlayColor}
      style={{ ...Styles.button, ...props.style.button }}
    >
      <View>{props.button || <Text>Navigate</Text>}</View>
    </TouchableHighlight>
  );

  return (
    <View style={{ ...Styles.container, ...props.style.container }}>
      <Image
        source={{
          uri: googleStaticMapUri(),
        }}
        style={computedDimensions}
      />
      {renderButton()}
    </View>
  );
}

Map.propTypes = propTypes;
Map.defaultProps = defaultProps;
