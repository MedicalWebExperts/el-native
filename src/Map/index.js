/* eslint no-console: ["error", { allow: ["error"] }] */
import React from 'react';
import PropTypes from 'prop-types';
import qs from 'qs';
import {
  View,
  Image,
  Text,
  Dimensions,
  Platform,
  TouchableHighlight,
  Linking,
} from 'react-native';

import Styles from './styles';

const propTypes = {
  latitude: PropTypes.number.isRequired,
  longitude: PropTypes.number.isRequired,
  width: PropTypes.number,
  height: PropTypes.number,
  apiKey: PropTypes.string.isRequired,
  maptype: PropTypes.string,
  zoom: PropTypes.number,
  onPress: PropTypes.func,
  style: PropTypes.object,
  button: PropTypes.element,
  underlayColor: PropTypes.string,
};

const defaultProps = {
  width: undefined,
  height: undefined,
  maptype: 'roadmap',
  zoom: 18,
  onPress: undefined,
  style: {},
  button: undefined,
  underlayColor: 'white',
};

export default function Map(props) {
  const dims = Dimensions.get('window');

  const computedDimensions = {
    width: Math.round(props.width || dims.width),
    height: Math.round(props.height || dims.width * (2 / 3)),
  };

  const googleStaticMapUri = () => {
    const baseUrl = 'https://maps.googleapis.com/maps/api/staticmap';
    const apiKey = props.apiKey;
    const params = {
      markers: [props.latitude, props.longitude].join(','),
      maptype: props.maptype,
      size: [computedDimensions.width, computedDimensions.height].join('x'),
      scale: 2,
      apiKey,
    };
    const query = qs.stringify(params, { arrayFormat: 'brackets' });
    const url = `${baseUrl}?${query}`;
    return url;
  };

  const handleOnPress = () => {
    const generateIos = () => `https://maps.apple.com/?q=${props.latitude},${props.longitude}&z=${props.zoom}`;
    const generateAndroid = () => `geo:${props.latitude},${props.longitude}`;

    const url = Platform.select({
      ios: generateIos,
      android: generateAndroid,
    })();

    Linking.canOpenURL(url).then((supported) => {
      if (!supported) {
        console.error(`Can't handle url: ${url}`);
        return;
      }
      Linking.openURL(url);
    }).catch(err => console.error('An error occurred', err));
  };

  const onPress = (props.onPress && typeof props.onPress === 'function') ? props.onPress : handleOnPress;

  const renderButton = () => (
    <TouchableHighlight
      onPress={onPress}
      underlayColor={props.underlayColor}
      style={{ ...Styles.button, ...props.style.button }}
    >
      <View>
        {props.button || <Text>Navigate</Text>}
      </View>
    </TouchableHighlight>
  );

  return (
    <View style={{ ...Styles.container, ...props.style.container }}>
      <Image
        source={{
          uri: googleStaticMapUri(),
        }}
        style={computedDimensions}
      />
      {renderButton()}
    </View>
  );
}

Map.propTypes = propTypes;
Map.defaultProps = defaultProps;
