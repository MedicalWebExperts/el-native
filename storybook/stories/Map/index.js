import React from 'react';
import { Text } from 'react-native';

import { storiesOf } from '@storybook/react-native';

import CenterView from '../CenterView';
import { Map } from '../../../src';
import colors from '../styles/colors';

const apiKey = process.env.ELNATIVE_GOOGLE_MAPS_APIKEY || '';

storiesOf('Map', module)
  .addDecorator(getStory => <CenterView>{getStory()}</CenterView>)
  .add('Default', () => <Map latitude={-34.5844583} longitude={-58.4230539} apiKey={apiKey} />)
  .add('Custom button', () => (
    <Map
      latitude={-34.5844583}
      longitude={-58.4230539}
      apiKey={apiKey}
      button={<Text>Custom</Text>}
    />
  ))
  .add('Custom button style', () => (
    <Map
      latitude={-34.5844583}
      longitude={-58.4230539}
      apiKey={apiKey}
      style={{
        button: {
          backgroundColor: colors.blueColor,
        },
      }}
    />
  ))
  .add('underlayColor', () => (
    <Map
      latitude={-34.5844583}
      longitude={-58.4230539}
      apiKey={apiKey}
      underlayColor={colors.primary}
    />
  ));
