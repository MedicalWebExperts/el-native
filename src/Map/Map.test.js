/* global describe it expect jest:true */
import React from 'react';
import { Text } from 'react-native';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

import Map from './index';

const apiKey = process.env.ELNATIVE_GOOGLE_MAPS_APIKEY || '';

const defaultProps = {
  latitude: -34.5844583,
  longitude: -58.4230539,
  apiKey,
};

describe('Snapshots', () => {
  it('should render default map', () => {
    const component = renderer.create(<Map
      {...defaultProps}
    />);
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });

  it('should render custom button', () => {
    const component = renderer.create(<Map
      {...defaultProps}
      button={<Text>custom</Text>}
    />);
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });

  it('should apply custom styles', () => {
    const component = renderer.create(<Map
      {...defaultProps}
      style={{
        container: { width: 50 },
        button: { backgroundColor: 'yellow' },
      }}
    />);
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });

  it('should apply underlayColor', () => {
    const component = renderer.create(<Map
      {...defaultProps}
      underlayColor="green"
    />);
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });
});

describe('Interactions', () => {
  it('should execute onPress callback', () => {
    const props = { ...defaultProps, onPress: jest.fn() };
    const wrapper = shallow(<Map {...props} />);

    wrapper.find('TouchableHighlight').prop('onPress')();
    expect(props.onPress).toHaveBeenCalledTimes(1);
  });
});
