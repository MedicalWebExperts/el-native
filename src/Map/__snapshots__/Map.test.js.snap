// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshots should apply custom styles 1`] = `
<View
  style={
    Object {
      "alignItems": "flex-end",
      "width": 50,
    }
  }
>
  <Image
    source={
      Object {
        "uri": "https://maps.googleapis.com/maps/api/staticmap?markers=-34.5844583%2C-58.4230539&maptype=roadmap&size=750x500&scale=2&apiKey=",
      }
    }
    style={
      Object {
        "height": 500,
        "width": 750,
      }
    }
  />
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "yellow",
        "bottom": 78,
        "padding": 30,
        "position": "relative",
      }
    }
  >
    <View
      style={null}
    >
      <Text>
        Navigate
      </Text>
    </View>
  </View>
</View>
`;

exports[`Snapshots should apply underlayColor 1`] = `
<View
  style={
    Object {
      "alignItems": "flex-end",
    }
  }
>
  <Image
    source={
      Object {
        "uri": "https://maps.googleapis.com/maps/api/staticmap?markers=-34.5844583%2C-58.4230539&maptype=roadmap&size=750x500&scale=2&apiKey=",
      }
    }
    style={
      Object {
        "height": 500,
        "width": 750,
      }
    }
  />
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "red",
        "bottom": 78,
        "padding": 30,
        "position": "relative",
      }
    }
  >
    <View
      style={null}
    >
      <Text>
        Navigate
      </Text>
    </View>
  </View>
</View>
`;

exports[`Snapshots should render custom button 1`] = `
<View
  style={
    Object {
      "alignItems": "flex-end",
    }
  }
>
  <Image
    source={
      Object {
        "uri": "https://maps.googleapis.com/maps/api/staticmap?markers=-34.5844583%2C-58.4230539&maptype=roadmap&size=750x500&scale=2&apiKey=",
      }
    }
    style={
      Object {
        "height": 500,
        "width": 750,
      }
    }
  />
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "red",
        "bottom": 78,
        "padding": 30,
        "position": "relative",
      }
    }
  >
    <View
      style={null}
    >
      <Text>
        custom
      </Text>
    </View>
  </View>
</View>
`;

exports[`Snapshots should render default map 1`] = `
<View
  style={
    Object {
      "alignItems": "flex-end",
    }
  }
>
  <Image
    source={
      Object {
        "uri": "https://maps.googleapis.com/maps/api/staticmap?markers=-34.5844583%2C-58.4230539&maptype=roadmap&size=750x500&scale=2&apiKey=",
      }
    }
    style={
      Object {
        "height": 500,
        "width": 750,
      }
    }
  />
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "red",
        "bottom": 78,
        "padding": 30,
        "position": "relative",
      }
    }
  >
    <View
      style={null}
    >
      <Text>
        Navigate
      </Text>
    </View>
  </View>
</View>
`;
